### ISSUE TITLE
Plugin / Pop dir questions

### ISSUE THREAD

**Issue by kalons7** [fman-community/fman#536]
1) What do the Pop dir commands do / how do they work?
2) Where are the default key bindings for these commands?  I don't seem them in 
AppData\Local\fman\Versions\1.3.2\Plugins\Core\Key Bindings.json or Key Bindings (Windows).json
3) What exactly is the mapping to commands for keyboard shortcuts?  The docs say "A list of all available commands can be found in the Core plugin"  I was able to create some by guessing, but there isn't anything like "create_directory" in the plugin code.  Is it some sort of LowerCase -> lower_case conversion on the class functions?

**Comment by mherrmann** [comment]
Hi,

 1. & 2. I don't know of a "Pop dir" command(?)
 3. Yes, there is a LowerCase -> lower_case conversion. It's an anachronism I'm not entirely happy with (see #104). It's "explained" in the docs [here](https://fman.io/docs/plugins-architecture).

**Comment by raguay** [comment]
There is a pop_dir command in the Favorites plugin for going back to a
previous directory. The current hotkey for one directory back is:

{ "keys": ["Ctrl+1"], "command":"pop_dir", "args": {"dirNum": 0}},

So, the default is the last directory is <ctrl>+1, two directories ago is
<ctrl>+2, etc. It goes to 9 directories in the past.

Richard

**Comment by raguay** [comment]
Just did an update to Favorites that fixed a buffer wrapping issue. Give it a try.

**Comment by kalons7** [comment]
Ah, that must be it.  Silly question: How do you update a plugin?

**Comment by mherrmann** [comment]
Uninstall and install it again. There's an issue for automating this, but it's not yet implemented I'm afraid.

**Comment by kalons7** [comment]
Thanks.

Locally, I've fixed the ExtendedStatusBar plugin to use as_human_readable, since it wasn't working, but I have a couple more questions.

4) Is there an event I can hook into when the active pane has changed (Tab), to refresh the status bar with that pane's info?
5) The plugin appears to implement 2 events on DirectoryPaneListener -- refresh and show_selected_files, but I don't see these documented at https://fman.io/docs/api#DirectoryPaneListener.  Is there more complete documentation somewhere?
6) What event can I use to refresh the status bar after the file selection changes?  show_selected_files seems to get called when a single file is toggled (Space) but not when extending a selection via Shift+Down Arrow, etc.  Or maybe there should be 2 status bars?

**Comment by mherrmann** [comment]
I'm afraid there are currently no events for when the active pane or the selected files change. See for instance [this comment](https://github.com/fman-users/fman/issues/292#issuecomment-360036718). The "active pane changed" event is #68. I don't think there's an issue yet for when the selected files change - feel free to open one. Finally, I think you'll need two status bars (#59).

`refresh` and `show_selected_files` have no special meaning for fman. They are general Python methods. The StatusBarExtended plugin [calls them](https://github.com/kek91/StatusBarExtended/blob/eb38c45499e5d5cc4d1f379c02c604d6df5c5d0a/statusbarextended/__init__.py#L99) itself.

**Comment by kalons7** [comment]
Where does the Favorites plugin store the list of favorite folders?  I want to reorder them.

**Comment by mherrmann** [comment]
I don't know but I'm sure you can figure this out from the [source code of the favorites plugin](https://github.com/raguay/favorites) ;-) If you see `load_/save_json('file.json')` somewhere then the favorites will be in `file.json` in your Settings plugin.

**Comment by kalons7** [comment]
Yes, I know.  I just thought the author of the plugin might reply and save me some time.

**Comment by raguay** [comment]
The directories are stored in your home directory titled `.favoritedirs`
and the shortcut directories are stored in `.shortenerdirs`. I put it there
so that my Alfred workflow and LaunchBar workflow can get to it easily. I
was working on workflows in Alfred and LaunchBar to allow the user to
reorder, bulk remove, etc their favorite directories. But, I never got
around to finishing them up. If you want them, I might be able to get
around to working on them next week some.

Richard

**Comment by kalons7** [comment]
I have no idea what Alfred and LaunchBar even are.  Thanks for the info, I can edit the files manually.

**Comment by raguay** [comment]
Alfred (https://www.alfredapp.com/) and LaunchBar (
https://www.obdev.at/products/launchbar/index.html) are application
launchers on macOS. Similar to Launchy on Windows 10 and Rolfi on Linux.
They allow you to create plugin/workflows/actions... to extend their
functionalities.
